src/xrpld/overlay/detail/PeerImp.cpp

```diff
@@ -2651,16 +2648,6 @@ PeerImp::onMessage(std::shared_ptr<protocol::TMSquelch> const& m)
     }
     PublicKey key(slice);
 
-    // Ignore non-validator squelch
```

Comment:
We don't expect this message with a non-validator key. Doesn't this open a door for spamming?

Reply:
Yes, the resource usage. We are currently charging a fee for a reason. Basically we don't expect this kind of message and will disconnect a peer if too many messages are received.

Reply:
This makes it complicated. So, I'm sending legitimate VP messages with a valid validator key to a peer. But the peer tells me to squelch messages from an invalid validator, which should not happen. If a peer doesn't want to see messages from me then I'll eventually disconnect this peer if I keep on receiving invalid squelch messages from that peer. The problem solved - the peer is not going to get any more messages from me.

Reply:
Do nothing. We don't want to reward a misbehaving peer by accepting its squelch request. We disconnect this peer and it will not receive messages from us.